runtime/src/rent_collector.rs

```diff
@@ -96,7 +96,7 @@ impl RentCollector {
                                 1
                             },
                     );
-                    account.lamports -= rent_due;
+                    account.checked_sub_lamports(rent_due).unwrap(); // will not fail. We check above.
```

Comment:
Seems like we can replace the pre-check with `if accounts.checked_sub_lamports(rent_due).is_some() {` ?

Reply:
The existing comparison is > instead of >=. checked_math will work fine on >=. The current code only works for >.
Was it a bug/oversight before that we had >= here?

Reply:
`                if account.lamports() > rent_due {
`

Reply:
Ah, right!  We don't want to change the data-deleting behavior.  Let's just leave it as you have it

Reply:
To avoid the unwrap you could do
```rust
let _ = account.checked_sub_lamports(rent_due); // will not fail. We check above.
```